src/session_impl.cpp

```diff
@@ -4705,7 +4705,7 @@ namespace {
 		{
 			std::shared_ptr<torrent_plugin> tp(e->new_torrent(
 				torrent_ptr->get_handle(), userdata));
-			if (tp) torrent_ptr->add_extension(std::move(tp));
```

Comment:
I think moving is correct here. even if `add_extension` doesn't take the shared pointer by value, the call site should move if it can. I also think that `add_extension()` probably *should* take it by value